knife-supermarket.gemspec

```diff
@@ -14,5 +14,5 @@ Gem::Specification.new do |s|
   s.homepage    = 'https://github.com/cwebberOps/knife-supermarket'
   s.license       = 'Apache 2.0'
   s.require_paths = ["lib"]
-  s.add_dependency "chef", ">= 0.10.10"
+  s.add_dependency "chef", ">= 0.10.10", "<= 12.11.18"
```

Comment:
you're in knife, you mostly want to hit stdout/stderr and don't care that much about logs

also the messaging you want probably looks more like:

```
if defined?(Chef::Knife::SupermarketDownload)
  puts "the knife-supermarket gem is deprecated and its features are in your version of chef-client, please uninstall the gem"
else
  puts "the knife-supermarket gem is deprecated and its features (and bugs) are frozen and have been moved into chef-client core, please upgrade to a newer version of chef-client"
  <old code here>
end
```